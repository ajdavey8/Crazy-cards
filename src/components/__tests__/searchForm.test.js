import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchForm from '../searchForm';

const handleChange = jest.fn();
const submitSearch = jest.fn();

const data = {
  title: 'Mr',
  firstName: 'Test',
  lastName: 'McTest',
  day: '17',
  month: '07',
  year: '1990',
  annualIncome: '12000',
  employmentStatus: 'Full Time',
  postcode: 'BN21 3RS'
};

describe('SearchForm', () => {
  it('renders and matches snapshot', () => {
    const tree = renderer
      .create(
        <SearchForm
          data={data}
          handleChange={handleChange}
          submitSearch={submitSearch}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
