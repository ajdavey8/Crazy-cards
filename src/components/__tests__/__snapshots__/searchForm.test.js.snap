// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders and matches snapshot 1`] = `
<form
  className="search-form"
  onSubmit={[MockFunction]}
>
  <div
    className="search-form__group"
  >
    <label
      className="search-form__label"
    >
      Title
    </label>
    <div
      className="search-form__radio-wrapper"
    >
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          defaultChecked={true}
          id="mr"
          name="title"
          onChange={[MockFunction]}
          type="radio"
          value="Mr"
        />
        <label
          className="search-form__radio-label"
          htmlFor="mr"
        >
          Mr
        </label>
      </div>
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="mrs"
          name="title"
          onChange={[MockFunction]}
          type="radio"
          value="Mrs"
        />
        <label
          className="search-form__radio-label"
          htmlFor="mrs"
        >
          Mr
        </label>
      </div>
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="ms"
          name="title"
          onChange={[MockFunction]}
          type="radio"
          value="Ms"
        />
        <label
          className="search-form__radio-label"
          htmlFor="ms"
        >
          Ms
        </label>
      </div>
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="miss"
          name="title"
          onChange={[MockFunction]}
          type="radio"
          value="Miss"
        />
        <label
          className="search-form__radio-label"
          htmlFor="miss"
        >
          Miss
        </label>
      </div>
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="dr"
          name="title"
          onChange={[MockFunction]}
          type="radio"
          value="Dr"
        />
        <label
          className="search-form__radio-label"
          htmlFor="dr"
        >
          Dr
        </label>
      </div>
    </div>
  </div>
  <div
    className="search-form__group"
  >
    <label
      className="search-form__label"
    >
      Name
    </label>
    <div
      className="search-form__input-wrapper"
    >
      <input
        className="search-form__text-input"
        name="firstName"
        onChange={[MockFunction]}
        placeholder="First Name"
        type="text"
        value="Test"
      />
      <input
        className="search-form__text-input"
        name="lastName"
        onChange={[MockFunction]}
        placeholder="Last Name"
        type="text"
        value="McTest"
      />
    </div>
  </div>
  <div
    className="search-form__group"
  >
    <label
      className="search-form__label"
    >
      Date of Birth
    </label>
    <div
      className="search-form__input-wrapper"
    >
      <input
        className="search-form__text-input"
        max="31"
        min="1"
        name="day"
        onChange={[MockFunction]}
        placeholder="DD"
        type="number"
        value="17"
      />
      <input
        className="search-form__text-input search-form__input-month"
        max="12"
        min="1"
        name="month"
        onChange={[MockFunction]}
        placeholder="MM"
        type="number"
        value="07"
      />
      <input
        className="search-form__text-input"
        max="2019"
        min="1900"
        name="year"
        onChange={[MockFunction]}
        placeholder="YYYY"
        type="number"
        value="1990"
      />
    </div>
  </div>
  <div
    className="search-form__group"
  >
    <label
      className="search-form__label"
    >
      What is your annual Income before tax?
    </label>
    <div
      className="search-form__input-wrapper"
    >
      <input
        className="search-form__text-input"
        name="annualIncome"
        onChange={[MockFunction]}
        placeholder="Annual Income"
        type="number"
        value="12000"
      />
    </div>
  </div>
  <div
    className="search-form__group"
  >
    <label
      className="search-form__label"
    >
      What is you employment status?
    </label>
    <div
      className="search-form__radio-wrapper"
    >
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="fullTime"
          name="employmentStatus"
          onChange={[MockFunction]}
          type="radio"
          value="fullTime"
        />
        <label
          className="search-form__radio-label"
          htmlFor="fullTime"
        >
          Full Time
        </label>
      </div>
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="partTime"
          name="employmentStatus"
          onChange={[MockFunction]}
          type="radio"
          value="partTime"
        />
        <label
          className="search-form__radio-label"
          htmlFor="partTime"
        >
          Part Time
        </label>
      </div>
      <div
        className="search-form__radio"
      >
        <input
          className="search-form__radio-input"
          id="student"
          name="employmentStatus"
          onChange={[MockFunction]}
          type="radio"
          value="student"
        />
        <label
          className="search-form__radio-label"
          htmlFor="student"
        >
          Student
        </label>
      </div>
    </div>
  </div>
  <div
    className="search-form__group"
  >
    <label
      className="search-form__label"
    >
      Postcode
    </label>
    <div
      className="search-form__input-wrapper"
    >
      <input
        className="search-form__text-input"
        name="postcode"
        onChange={[MockFunction]}
        placeholder="Postcode"
        type="text"
        value="BN21 3RS"
      />
    </div>
  </div>
  <div>
    <button
      className="search-form__button"
    >
      Get Results
    </button>
  </div>
</form>
`;
